src/main/java/com/minecolonies/coremod/entity/ai/citizen/beekeeper/EntityAIWorkBeekeeper.java

```diff
@@ -433,6 +424,39 @@ public boolean walkingToAnimal(final AnimalEntity animal)
         }
     }
 
+    /**
+     * Determines if the beekeeper is ready to start breeding.
+     *
+     * @return whether the beekeeper is ready to start breeding.
+     */
+    private boolean isReadyForBreeding()
+    {
+        final ItemListModule flowersModule = getOwnBuilding().getModuleMatching(ItemListModule.class, m -> m.getId().equals(BUILDING_FLOWER_LIST));
+        final List<BeeEntity> bees = new ArrayList<>(searchForAnimals(world, getOwnBuilding()));
+
+        final int breedableAnimals = (int) bees.stream().filter(animal -> animal.getAge() == 0).count();
+
+        boolean canBreed = getOwnBuilding().getSetting(BuildingBeekeeper.BREEDING).getValue() && !hasMaxAnimals(bees) && breedableAnimals >= NUM_OF_ANIMALS_TO_BREED;
```

Comment:
Maybe we use the setting here as a early exit before counting the animals (as that is a costly check).